lib/IRGen/ESTreeIRGen.cpp

```diff
@@ -619,12 +642,25 @@ void ESTreeIRGen::emitIteratorCloseSlow(
     emitTryCatchScaffolding(
         noReturn,
         // emitBody.
-        [this, returnMethod, &iteratorRecord](BasicBlock * /*catchBlock*/) {
-          Builder.createCallInst(
-              returnMethod,
-              /* newTarget */ Builder.getLiteralUndefined(),
-              iteratorRecord.iterator,
-              {});
+        [this, returnMethod, &iteratorRecord, isAsyncIterator, astNode](BasicBlock * catchBlock) {
+          SurroundingTry thisTry{
+            curFunction(),
+            astNode,
+            catchBlock,
+            {},
+            [this, &iteratorRecord, isAsyncIterator, returnMethod](
+                ESTree::Node *,
+                ControlFlowChange cfc,
+                BasicBlock *continueTarget) {
+              auto *callResult = Builder.createCallInst(
```

Comment:
Why was this moved into the finalizer instead of left in the body block like it used to be?